caikit/runtime/interceptors/caikit_runtime_server_wrapper.py

```diff
@@ -191,9 +192,14 @@ def __init__(self, method):
                     # Get the original grpc.RpcMethodHandler for this RPC method
                     original_rpc_handler = handler.service(DummyHandlerCallDetails(fqm))
 
-                    # Make sure this is a unary-unary RPC
-                    if not original_rpc_handler.unary_unary:
-                        raise NotImplementedError("Unary-unary RPCs only!")
+                    # Make sure this is a supported RPC flavor
```

Comment:
This line is pretty confusing to me as I'd expect the logic to be exactly the opposite of what you have written, but I've seen it working in action, so I'll trust I'm just misreading it!

Reply:
🤔 
It raises if neither `unary_unary` nor `unary_stream` are set, and those are the only two flavors we support. This should be equivalent to
```
if original_rpc_handler.stream_unary or original_rpc_handler.stream_stream:
```

Reply:
🤦 I'm... confused. Carry on!